plugins/modules/net_tools/ldap/ldap_attrs.py

```diff
@@ -264,7 +265,8 @@ def exact(self):
     def _is_value_present(self, name, value):
         """ True if the target attribute has the given value. """
         try:
-            filterstr = "(%s=%s)" % (name, value.decode())
+            escaped_value = ldap.filter.escape_filter_chars(value)
+            filterstr = "(%s=%s)" % (name, escaped_value.decode())
```

Comment:
Do you need to specify a character encoding for `decode()`, or does `escape_filer_chars` always return US ASCII no matter what the input was?

Reply:
You are right, the `value` is converted using `ansible.module_utils.common.text.converters.to_bytes` in the [`_normalize_values`](https://github.com/ansible-collections/community.general/blob/fa924aae31c1675bcf4adb2a9ac56711af71f5eb/plugins/modules/net_tools/ldap/ldap_attrs.py#L206) function which means that I should probably decode this value before passing it to the `ldap.filter.escape_filter_chars` function.

The behavior seems to vary between python versions. When I tested this on a Centos7 machine with python 2.7.5 it was working:

```python
>>> print (sys.version)
2.7.5 (default, Jun 28 2022, 15:30:04) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
>>> import ldap.filter
>>> value = 'test'.encode('utf-8')
>>> escaped_value = ldap.filter.escape_filter_chars(value)
>>> escaped_value
'test'
```
But with Python `3.10.7` I get a `TypeError`:

```python
>>> print (sys.version)
3.10.7 (main, Sep  7 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-1)]
>>> import ldap.filter
>>> value = 'test'.encode('utf-8')
>>> escaped_value = ldap.filter.escape_filter_chars(value)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rkupferschmid/.local/share/virtualenvs/ansible-puzzle-UgRIbmN5/lib64/python3.10/site-packages/ldap/filter.py", line 41, in escape_filter_chars
    s = assertion_value.replace('\\', r'\5c')
TypeError: a bytes-like object is required, not 'str'